File: mil/src/org/zkoss/mil/Textbox.java
/* Textbox.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		May 25, 2007 9:49:27 AM, Created by henrichen
}}IS_NOTE

Copyright (C) 2007 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 2.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/

package org.zkoss.mil;

import org.zkoss.mil.impl.InputElement;
import org.zkoss.zk.ui.WrongValueException;

/**
 * A generic text box.
 * 
 * @author henrichen
 */
public class Textbox extends InputElement {
	private static final long serialVersionUID = 200705251351L;
	private String _type = "text";

	public Textbox() {
	}
	public Textbox(String value) throws WrongValueException {
		setValue(value);
	}
	public Textbox(String label, String value) throws WrongValueException {
		setValue(value);
	}

	/** Returns the value.
	 * The same as {@link #getText}.
	 * <p>Default: "".
	 * @exception WrongValueException if user entered a wrong value
	 */
	public String getValue() throws WrongValueException {
		return getText();
	}
	/** Sets the value.
	 *
	 * @param value the value; If null, it is considered as empty.
	 * @exception WrongValueException if value is wrong
	 */
	public void setValue(String value) throws WrongValueException {
		setText(value);
	}

	//-- super --//
	/** Coerces the value passed to {@link #setValue}.
	 *
	 * <p>Default: convert null to an empty string.
	 */
	protected Object coerceFromString(String value) throws WrongValueException {
		return value != null ? value: "";
	}
	/** Coerces the value passed to {@link #setValue}.
	 *
	 * <p>Default: convert null to an empty string.
	 */
	protected String coerceToString(Object value) {
		return value != null ? (String)value: "";
	}

	/** random text */
	protected int getInternalType() {
		return ANY;
	}
	
	/** Returns the type.
	 * <p>Default: text.
	 */
	public String getType() {
		return _type;
	}
	/** Sets the type.
	 * @param type the type. Acceptable values are "text" and "password".
	 */
	public void setType(String type) throws WrongValueException {
		if (!"text".equals(type) && !"password".equals(type))
			throw new WrongValueException("Illegal type: "+type);

		if (!_type.equals(type)) {
			_type = type;
			smartUpdateConstraints();
		}
	}
}

File: zul/src/archive/web/js/zul/inp/mold/textbox.js
/* textbox.js

	Purpose:
		
	Description:
		
	History:
		Sat Dec 13 23:32:20     2008, Created by tomyeh

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 3.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	var zcls = this.getZclass(),
		uuid = this.uuid,
		isRounded = this.inRoundedMold();
	if(this.isMultiline()) 
		out.push('<textarea', this.domAttrs_(), '>', this._areaText(), '</textarea>');
	else if(!isRounded)
		out.push('<input', this.domAttrs_(), '/>');
	else {
		out.push('<i', this.domAttrs_({text:true}), '>',
				'<input id="', uuid, '-real"', 'class="', zcls, '-inp"', 
				this.textAttrs_(), '/>', '<i id="', uuid, '-right-edge"',
				'class="', zcls, '-right-edge');
		/*
		if (this._readonly)
			out.push(' ', zcls, ' -right-edge-readonly');
		*/
		out.push('"></i></i>');
	}
}